test/jdk/javax/imageio/plugins/jpeg/WriteJPEGThumbnailTest.java

```diff
@@ -0,0 +1,160 @@
+        if (expected != observed) {
+            throw new Error("expected " + expected + ", but observed " + observed);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        // this always passed. 21800 * 3 = 65400, which fits in a 65535-byte segment.
+        boolean b1 = new WriteJPEGThumbnailTest(100, 218).run();
+
+        // this failed prior to resolving 8351110. 21900 * 3 = 65700, which is too large
+        // for a JPEG segment. Now we clip the thumbnail to make it fit. (Previously
+        // we wrote a corrupt JPEG file.)
+        boolean b2 = new WriteJPEGThumbnailTest(100, 219).run();
+        if (!(b1 && b2)) {
+            System.err.println("Test failed");
+            throw new Error("Test failed");
+        }
+    }
+
+    final int thumbWidth, thumbHeight;
+
+    public WriteJPEGThumbnailTest(int thumbWidth, int thumbHeight) {
+        this.thumbWidth = thumbWidth;
+        this.thumbHeight = thumbHeight;
+    }
+
+    public boolean run() throws Exception {
+        System.out.println("Testing thumbnail " + thumbWidth + "x" + thumbHeight + "...");
+        try {
+            byte[] jpegData;
+            BufferedImage thumbnail;
+            try (ByteArrayOutputStream byteOut = new ByteArrayOutputStream()) {
+                try {
+                    thumbnail = writeImage(byteOut);
+                } finally {
+                    jpegData = byteOut.toByteArray();
+                }
+            }
+
+            ImageReader reader = getJPEGImageReader();
+            ImageInputStream stream = ImageIO.createImageInputStream(new ByteArrayInputStream(jpegData));
+            reader.setInput(stream);
+            assertEquals(1, reader.getNumThumbnails(0));
+
+            // we may have a subset of our original thumbnail, that's OK
+            BufferedImage readThumbnail = reader.readThumbnail(0, 0);
+            for (int y = 0; y < readThumbnail.getHeight(); y++) {
+                for (int x = 0; x < readThumbnail.getWidth(); x++) {
+                    int rgb1 = thumbnail.getRGB(x, y);
+                    int rgb2 = readThumbnail.getRGB(x, y);
+                    assertEquals(rgb1, rgb2);
+                }
+            }
+            System.out.println("\tTest passed");
+        } catch (Throwable e) {
+            e.printStackTrace();
+            return false;
+        }
+        return true;
+    }
+
+    private BufferedImage writeImage(OutputStream out) throws IOException {
+        BufferedImage thumbnail = createImage(thumbWidth, thumbHeight);
+        BufferedImage bi = createImage(thumbnail.getWidth() * 10, thumbnail.getHeight() * 10);
+
+        ImageWriter writer = ImageIO.getImageWritersByFormatName("jpeg").next();
+
+        try (ImageOutputStream outputStream = ImageIO.createImageOutputStream(out)) {
+            writer.setOutput(outputStream);
+
+            // Write the main image
+            IIOImage img = new javax.imageio.IIOImage(bi, List.of(thumbnail), null);
```

Comment:
```suggestion
            IIOImage img = new IIOImage(bi, List.of(thumbnail), null);
```
The package name on right side of the assignment is redundant.